OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -269,33 +269,34 @@ public string VoicePhraseForOrder(Actor self, Order order)
 		public bool HasSpace(int weight) { return totalWeight + reservedWeight + weight <= Info.MaxWeight; }
 		public bool IsEmpty(Actor self) { return cargo.Count == 0; }
 
-		public Actor Peek(Actor self) { return cargo.Peek(); }
+		public Actor Peek(Actor self) { return cargo.Last(); }
 
-		public Actor Unload(Actor self)
+		public Actor Unload(Actor self, Actor passenger = null)
 		{
-			var a = cargo.Pop();
+			passenger = passenger ?? cargo.Last();
+			cargo.Remove(passenger);
```

Comment:
You removed the logic to handle if the passenger wasn't in the transport from my example, but that was the most important part!

The logic in the rest of the method is bogus if `cargoRemove(passenger)` returns false.

Reply:
Sorry, somehow missed that part, pushed a fix